go-controller/pkg/node/gateway_udn.go

```diff
@@ -975,3 +984,56 @@ func (udng *UserDefinedNetworkGateway) removeDefaultRouteFromVRF() error {
 	}
 	return nil
 }
+
+// updateAdvertisedUDNIsolationRules adds nft rules to drop locally generated traffic destined to the UDN subnets for
+// BGP advertised networks. It blocks access to the full UDN subnet to handle a case in L3 when a node tries to access
+// a host subnet available on a different node. When isNetworkAdvertised is false the rules are removed.
+// Example:
+//
+//	chain udn-bgp-drop {
+//	  comment "Drop traffic generated locally towards the advertised UDN subnets"
+//	  type filter hook output priority filter; policy accept;
+//	  ip daddr 10.10.0.0/16 counter packets 0 bytes 0 drop comment "cluster_udn_l3network"
+//	}
+func (udng *UserDefinedNetworkGateway) updateAdvertisedUDNIsolationRules(isNetworkAdvertised bool) error {
+	nft, err := nodenft.GetNFTablesHelper()
+	if err != nil {
+		return err
+	}
+	tx := nft.NewTransaction()
+
+	if !isNetworkAdvertised {
```

Comment:
do we have a unit test for this logic?
add the rule, trigger an unmatch update and ensure rule is removed logic?

Reply:
Added unit tests!